app/controllers/restaurants_controller.rb

```diff
@@ -2,47 +2,41 @@ class RestaurantsController < ApplicationController
   skip_before_action :verify_authenticity_token
 
   def index
-    Restaurant.sorted
+    render json: { restaurants: Restaurant.sorted_by_name.map{|r| RestaurantSerializer.new(r)} }
   end
 
   def show
-    Restaurant.find(params[:id])
+    render json: { restaurant: RestaurantSerializer.new(find_restaurant(params[:id])) }
   end
 
   def new
-    Restaurant.new
+    render json: { restaurant: RestaurantSerializer.new(Restaurant.new) }
   end
 
   def create
-    Restaurant.create(restaurants_params)
+    render json: { restaurant: RestaurantSerializer.new(Restaurant.create(restaurants_params)) }
   end
 
   def edit
-    Restaurant.find(params[:id])
+    render json: { restaurant: RestaurantSerializer.new(find_restaurant(params[:id])) }
   end
 
   def update
-    restaurant = Restaurant.find(params[:id])
-
-    if restaurant
-      restaurant.update_attributes(restaurants_params)
-    end
-  end
-
-  def delete
-    Restaurant.find(params[:id])
+    restaurant = find_restaurant(params[:id])
+    restaurant.update_attributes(restaurants_params) if restaurant
   end
 
   def destroy
-    restaurant = Restaurant.find(params[:id])
-
-    if restaurant
-      restaurant.destroy
-    end
+    restaurant = find_restaurant(params[:id])
+    restaurant.destroy if restaurant
   end
 
   private
   def restaurants_params
-    params.require(:restaurant).permit(:name, :cuisine, :rating, :is_10_bis, :address, :max_delivery_time_in_minutes)
+    params.require(:restaurant).permit(:name, :cuisine, :is_10_bis, :address, :max_delivery_time_in_minutes)
+  end
+
+  def find_restaurant(id)
```

Comment:
a more "railsy" way to do it might be:
```
  before_action :set_restaurant, only: %i[index new create]

....

  def set_restaurant
    @restaurant = Restaurant.find(params[:restaurant_id])
  end
```

then you can just access `@restaurant`